library/asb_get_state.py

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/python
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.0',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+import os
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    from kubernetes import client, config
+    from kubernetes.client.rest import ApiException
+    config.load_kube_config()
+    api = client.CoreV1Api()
+except Exception as error:
+    ansible_module.fail_json(
+        msg="Error attempting to load kubernetes client: {}".format(error))
+
+
+ENV_NAME = "POD_NAME"
+ENV_NAMESPACE = "POD_NAMESPACE"
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            key=dict(required=True, type='str'),
+        ),
+    )
+
+    try:
+        name = os.environ[ENV_NAME]
+        namespace = os.environ[ENV_NAMESPACE]
+    except Exception as e:
+        module.fail_json(
+            msg="Error attempting to get name/namespace from environment: {}".format(e))
+
+    try:
+        config_map = api.read_namespaced_config_map(name, namespace)
```

Comment:
@PhilipGough This actually needs to read it from disk as the configmap will be mounted into the pod at execution time. I am thinking of mounting it at /etc/apb/state

Reply:
@maleck13 ah ok, makes sense that it would be mounted as a volume sorry.